src/objectExplorer/objectExplorerService.ts

```diff
@@ -257,19 +257,24 @@ export class ObjectExplorerService {
             // retrieve saved connections first when opening object explorer
             // for the first time
             let savedConnections = this._connectionManager.connectionStore.loadAllConnections();
+            // if OE doesn't exist or if there was a change in saved connections
+            // then build the nodes off of saved connections
             if ((!this._objectExplorerProvider.objectExplorerExists ||
-                savedConnections.length !== this._rootTreeNodeArray.length) &&
-                savedConnections.length > 0) {
-                this._rootTreeNodeArray = [];
-                this.getSavedConnections();
-                this._objectExplorerProvider.objectExplorerExists = true;
-                return this.sortByServerName(this._rootTreeNodeArray);
-            } else {
-                if (this._rootTreeNodeArray.length > 0) {
+                savedConnections.length !== this._rootTreeNodeArray.length)) {
```

Comment:
>savedConnections.length !== this._rootTreeNodeArray.length [](start = 16, length = 58)

What about if there's the same number of saved connections but they were changed? You should probably be doing a deeper check here to verify that.

Reply:
I will handle that when fixing this - https://github.com/microsoft/vscode-mssql/issues/1346
since editing connections or manually adding connections needs an event for workspace settings change